client/client.go

```diff
@@ -171,17 +202,10 @@ func WithHost(host string) func(*Client) error {
 		c.proto = hostURL.Scheme
 		c.addr = hostURL.Host
 		c.basePath = hostURL.Path
-		if c.client == nil {
-			client, err := defaultHTTPClient(host)
-			if err != nil {
-				return err
-			}
-			return WithHTTPClient(client)(c)
-		}
```

Comment:
I removed this because `NewClientWithOpts` always sets a default `c.client`, so there is no way for this branch to ever be hit. This is related to the fix from #36270. Before that fix we always set a new defaultHTTPClient, but now we never set a client because `c.client` is already set.  cc @dperny